File: Samples/Win7Samples/multimedia/gdi/wcs/Utilities.cpp
//+--------------------------------------------------------------------------
//
// THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF
// ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO
// THE IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
// PARTICULAR PURPOSE.
//
// Copyright (c) Microsoft Corporation. All rights reserved.
//
//  Abstract:
//      Contains definitions of  functions not specific to the color sample
//
//----------------------------------------------------------------------------

#include "precomp.h"
#include "Utilities.h"

//+---------------------------------------------------------------------------
//
//  Function:
//      SafeULongMult
//
//  Synopsis:
//      This function multiplies two ULONG numbers together,
//      and returns E_FAIL if numeric overflow occurs.
//
//----------------------------------------------------------------------------
HRESULT SafeULongMult(
    __in ULONG ulMultiplicand,
    __in ULONG ulMultiplier,
    __out ULONG *pulResult
    )
{
    ULONG64 ul64Multiplicand = ulMultiplicand;
    ULONG64 ul64Multiplier = ulMultiplier;
    ULONG64 Product = ul64Multiplicand*ul64Multiplier;
    if (Product <= ((ULONG)-1))
    {
        *pulResult = (ULONG)Product;
        return S_OK;
    }
    return E_FAIL;
}

File: Samples/Win7Samples/netds/messagequeuing/LargeMessageQueue/CS/TestLargeMessageQueue/Utilities.cs
// --------------------------------------------------------------------
//
//  Copyright (c) Microsoft Corporation.  All rights reserved
//
// THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF
// ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO
// THE IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
// PARTICULAR PURPOSE.
//
// -------------------------------------------------------------------- 
namespace TestLargeMessageQueue
{
    using System;
    using System.Windows;
    using System.Windows.Controls;
    using System.Windows.Media;

    class Utilities
    {
        private Utilities()
        {
        }

        // Start Indicator for operation success/failure
        internal static void StartProgressBar(ProgressBar progressBar)
        {
            progressBar.Visibility = Visibility.Visible;
            progressBar.Foreground = Brushes.Green;
        }

        // Stop Indicator on operation completion
        internal static void StopProgressBar(ProgressBar progressBar)
        {
            progressBar.Value = progressBar.Maximum;
        }

        // Change Indicator on operation failure
        internal static void ErrorProgressBar(ProgressBar progressBar)
        {
            progressBar.Foreground = Brushes.Red;
            progressBar.Value = progressBar.Maximum;
        }
    }
}
